unit_test.py

```diff
@@ -0,0 +1,1371 @@
+# This is file contains a unit test suit which completely test the backend
+# functions of our app.
+# Note:
+# 1.We create some random functions before the test suit to add randomness to
+# our tests.
+# 2.Some later tests use some of the previous test to avoid code repetitions.
```

Comment:
DRYness is NOT a virtue in testing. You are actually better off repeating yourself.